contracts/XorShift128Plus.cdc

```diff
@@ -0,0 +1,95 @@
+import Crypto
+
+/// Defines a xorsift128+ pseudo random generator as a resource
+///
+access(all) contract XorShift128Plus {
+
+    access(all) let StoragePath: StoragePath
+    access(all) let PublicPath: PublicPath
+
+    /// While not limited to 128 bits of state, this PRG is largely informed by XORShift128+
+    ///
+    access(all) resource PRG {
+
+        // The states below are of type Word64 (instead of UInt64) to prevent overflow/underflow as state evolves
+        //
+        access(all) var state0: Word64
+        access(all) var state1: Word64
+
+        init(sourceOfRandomness: [UInt8]) {
+            pre {
+                sourceOfRandomness.length == 32: "Expecting 32 bytes as sourceOfRandomness"
```

Comment:
This should be 16 as the xorshift128+ requires. 

```suggestion
                sourceOfRandomness.length == 16: "Expecting 16 bytes as input seed"
```

Reply:
Since PRG is now a struct, it's available to construct by its initializer. So I've updated the `init()` method to condition `sourceOfRandomness.length >= 16` and consolidated `createPRG()`'s implementation to PRG's init().